htlcswitch/link.go

```diff
@@ -2559,10 +2585,28 @@ func (l *channelLink) drivePendingStfuSends() {
 				handleErr(err)
 				return
 			}
+			l.tryResolveQuiescenceRequests()
 		})
 	}
 }
 
+// tryResolveQuiescenceRequests checks whether the quiescence protocol has
+// successfully completed. If it has, it completes the outstanding asynchronous
+// request, if there is one.
+func (l *channelLink) tryResolveQuiescenceRequests() {
+	if l.quiescer.isQuiescent() {
+		if resp := l.activeQuiescenceRequest; resp != nil {
```

Comment:
Def need to circle back to this section, but my gut instinct is that I think we can further encapsulate this logic into the `quiescer` itself. As here we're keeping track of the active requests, and also state of the `quiescer` vs just asking it to advance to the next state if it's able to do so.